packages/flutter/lib/src/material/button.dart

```diff
@@ -343,9 +353,8 @@ class _MaterialButtonState extends State<MaterialButton> {
       constraints: new BoxConstraints(
         minWidth: widget.minWidth ?? buttonTheme.minWidth,
         minHeight: height,
-        maxHeight: height
       ),
-      child: contents
+      child: new IntrinsicHeight(child: contents),
```

Comment:
we can't do this with IntrinsicHeight, that's too expensive

Reply:
Yeah, I wondered if that might be the case.  I had three choices:

1) Use IntrinsicHeight.  It might be slow, but the vast majority of uses are n=1 (a single Text widget), so I didn't think the n² would kick in very often.
2) Use a Column with MajorAxisSize.min.  This works, but it breaks things.  In particular the sector demo's tests broke when I did this, and I took that as an indication that customer's code might also break.
3) Use a really ugly hack to detect a Text widget-like interface and special case them, multiplying the fixed height by their scale factor (definitely not a clean choice).

If you're cool with breaking some code, I'm happy to implement choice 2.  It results in the right behavior, and also communicates the fact that the height *can* change to the developers.